### PR TITLE
docs: allow for serving 2 specifications

### PR DESCRIPTION
**Pull request by lag-of-death**
## Checklist

- [ ] Tests have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## What kind of change does this PR introduce?

_feature_

## What is the current behavior? What is the new behavior?

This PR allows for multiple (atm just 2) OAS documents to be served by Prism.

## Does this PR introduce a breaking change?

No.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - The CLI will now propose operation paths with meaningful examples [#671](https://github.com/stoplightio/prism/pull/671)
 - Prism reloads itself every time there are changes being made to the specified document [#689](https://github.com/stoplightio/prism/pull/689)
 - The Test Harness framework now requires the `${document}` parameter explicitly [#720](https://github.com/stoplightio/prism/pull/720)
+- Prism can serve two OAS documents at the same time [#730](https://github.com/stoplightio/prism/pull/730)
```
Comment (XVincentX) [inline CHANGELOG.md]: We do not need a changelog line.

File: nginx.conf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  server {
+    listen 8080;
+
+    location /app_1 {
+      proxy_pass http://prism_1:4010/;
+    }
+
+    location /app_1/ {
+      proxy_pass http://prism_1:4010/;
+    }
```
Comment (XVincentX) [inline nginx.conf]: I think you can put both in the same line, without repeating yourself.

File: nginx.conf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    location /app_2 {
+      proxy_pass http://prism_2:4010/;
+    }
+
+    location /app_2/ {
+      proxy_pass http://prism_2:4010/;
+    }
```
Comment (XVincentX) [inline nginx.conf]: dtto

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf
+    ports:
+      - ${PRISM_PROXY_PORT}:8080
+    depends_on:
+      - prism_1
+      - prism_2
+  prism_1:
+    image: stoplight/prism:3
+    command: mock --cors -p 4010 --host 0.0.0.0 /usr/data/doc1.yaml
```
Comment (XVincentX) [inline docker-compose.yaml]: Let's put a remote document, so people can try out this document out of the box

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      - ${PRISM_DOC_1}:/usr/data/doc1.yaml
+    expose:
+      - 4010
+  prism_2:
+    image: stoplight/prism:3
+    command: mock --cors -p 4010 --host 0.0.0.0 /usr/data/doc2.yaml
+    volumes:
+      - ${PRISM_DOC_2}:/usr/data/doc2.yaml
+    expose:
+      - 4010
```
Comment (XVincentX) [inline docker-compose.yaml]: This is already declared in the docker file and it's not required.

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    command: mock --cors -p 4010 --host 0.0.0.0 /usr/data/doc1.yaml
+    volumes:
+      - ${PRISM_DOC_1}:/usr/data/doc1.yaml
+    expose:
+      - 4010
+  prism_2:
+    image: stoplight/prism:3
+    command: mock --cors -p 4010 --host 0.0.0.0 /usr/data/doc2.yaml
+    volumes:
+      - ${PRISM_DOC_2}:/usr/data/doc2.yaml
```
Comment (XVincentX) [inline docker-compose.yaml]: Let's use a remote document so testing this out is easier

File: docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+version: "3"
+services:
+  proxy:
+    image: nginx:1.17.4
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf
+    ports:
+      - ${PRISM_PROXY_PORT}:8080
```
Comment (XVincentX) [inline docker-compose.yaml]: 8080 is just fine. Again this is an example we're going to put in the documentation, not a real feature
```suggestion
      - 8080:8080
```

File: docker-compose.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+version: "3"
+services:
+  proxy:
+    image: nginx:1.17.4
```
Comment (XVincentX) [inline docker-compose.yaml]:
```suggestion
    image: nginx:alpine
```

File: docker-compose.yaml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf
+    ports:
+      - 8080:8080
```
Comment (XVincentX) [inline docker-compose.yaml]: You can listen on the 80 by default

```suggestion
      - 8080:80
```

File: docs/guides/multiple-documents.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
+# Serving multiple OAS documents
+
+A single Prism instance serves one OAS document.
+In order to serve multiple documents, you can use `docker-compose` utility.
```
Comment (XVincentX) [inline docs/guides/multiple-documents.md]:
```suggestion
In order to serve multiple documents, the best way is to usually have multiple instances of Prism running on different documents and then serve their content using a reverse Proxy. We're going to show an example using multiple docker containers.
```

File: docs/guides/multiple-documents.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf
+    ports:
+      - "8080:80"
+    depends_on:
+      - prism_1
+      - prism_2
+  prism_1:
+    image: stoplight/prism:3
+    command: mock -p 4010 --host 0.0.0.0 https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v2.0/yaml/petstore.yaml
```
Comment (XVincentX) [inline docs/guides/multiple-documents.md]: Thes urls are too long. Any idea what we could do here?

File: docs/guides/multiple-documents.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+      proxy_pass http://prism_1:4010/;
+    }
+
+    location /app_2 {
+      rewrite ^/app_2/(.*) /$1 break;
+      proxy_pass http://prism_2:4010/;
+    }
+  }
+}
+```
```
Comment (XVincentX) [inline docs/guides/multiple-documents.md]: We need to put a note saying this Nginx file is not complete; this is missing different required things when proxying stuff such as IP address re-write, Forward headers and stuff.

File: docs/guides/multiple-documents.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+This configuration will allow access to the first Prism instance on `localhost:8080/app_1` and the second instance on `localhost:8080/app_2` endpoint.
+
+## Running the cluster
+
+Let's first place the aforementioned files under the same directory - here `prism`.
+
+```
+prism
```
Comment (XVincentX) [inline docs/guides/multiple-documents.md]: Let's use another name to not confuse the user with the source code
```suggestion
prism-multi-example
```

File: docs/guides/multiple-documents.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+```
+1. http://localhost:8080/app_1
+2. http://localhost:8080/app_2
+```
+
+You can now try these out by issuing:
+
+```
+1. curl localhost:8080/app_1/pets
+2. curl localhost:8080/app_2/pets
```
Comment (XVincentX) [inline docs/guides/multiple-documents.md]: If I recall correctly the OAS2 document serves `pet/` while the OAS3 serves `pets`

File: docs/guides/multiple-documents.md (hunk 11 above)
Comment (lag-of-death) [reply docs/guides/multiple-documents.md]: @XVincentX, we could go back to envs, no other ideas

File: docs/guides/multiple-documents.md (hunk 14 above)
Comment (lag-of-death) [reply docs/guides/multiple-documents.md]: don't know, but these 2 will consistently list pets

File: docs/guides/multiple-documents.md (hunk 12 above)
Comment (lag-of-death) [reply docs/guides/multiple-documents.md]: https://github.com/stoplightio/prism/pull/730/files#diff-dcfe973ffec48a8862b3e311c8f6571bR57

File: docs/guides/multiple-documents.md (hunk 14 above)
Comment (XVincentX) [reply docs/guides/multiple-documents.md]: You're right.

File: docs/guides/multiple-documents.md (hunk 11 above)
Comment (XVincentX) [reply docs/guides/multiple-documents.md]: Let's not worry about this

**Pull request merged**